proptest/src/arbitrary/_std/path.rs

```diff
@@ -9,15 +9,72 @@
 
 //! Arbitrary implementations for `std::path`.
 
-use std::path::*;
+use std::{ffi::OsString, path::*};
 
-// TODO: Figure out PathBuf and then Box/Rc/Box<Path>.
+use crate::{
+    arbitrary::{SMapped, StrategyFor},
+    prelude::{any, any_with, Arbitrary, Strategy},
+    std_facade::{string::ToString, Arc, Box, Rc, Vec},
+    strategy::{statics::static_map, MapInto},
+};
 
 arbitrary!(StripPrefixError; Path::new("").strip_prefix("a").unwrap_err());
 
+/// This implementation takes two parameters: a range for the number of components, and a regex to
+/// determine the string. It generates either a relative or an absolute path with equal probability.
+///
+/// Currently, this implementation does not generate:
+///
+/// * Paths that are not valid UTF-8 (this is unlikely to change)
+/// * Paths with a [`PrefixComponent`](std::path::PrefixComponent) on Windows, e.g. `C:\` (this may
+///   change in the future)
+impl Arbitrary for PathBuf {
+    type Parameters = <Vec<OsString> as Arbitrary>::Parameters;
+    type Strategy = SMapped<(bool, Vec<OsString>), Self>;
+
+    fn arbitrary_with(args: Self::Parameters) -> Self::Strategy {
+        static_map(
+            (any::<bool>(), any_with::<Vec<OsString>>(args)),
```

Comment:
i'm not familiar with the `OsString` arbitrary impl but I'm wondering if it can generate a string with a delimiter in it. will pull this locally and mess around more. otherwise this looks good.

Reply:
OsString strategy does indeed generate strings with the delimiter.

Reply:
I think that's okay, the number of delimiters is not a hard and fast rule here.

I'm more concerned about the number of components being too large by default.

I'm wondering if we should define a PathParameters struct with a custom default value for the number of delimiters.

Reply:
I did notice that quite a few components were getting generated. I think a `PathParameters` struct with a default number and ensuring the output has that number (which may involve trimming any excess generated) could make sense.  I think it'd be an un-intuitive if a user asks for `n` segments and we give them `n` + `m`.

Reply:
That makes sense. You're welcome to work on it, or I'll get around to it at some point later this week (speaking at RustConf in 3 days 😅)

Reply:
@matthew-russo OK, done. Let me know if this makes sense.

Reply:
Sorry work and personal life have kept me quite busy so I missed your last message. Thanks for getting back to this and congrats on the talk!